src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -856,20 +945,35 @@ codet java_bytecode_convert_methodt::convert_instructions(
       const dereference_exprt deref(pointer, pointer.type().subtype());
 
       const member_exprt data_ptr(
-        deref, "data", pointer_typet(java_type_from_char(type_char)));
+        deref,
+        "data",
+        pointer_typet(java_type_from_char(type_char)));
 
       plus_exprt data_plus_offset(data_ptr, op[1], data_ptr.type());
       typet element_type=data_ptr.type().subtype();
       const dereference_exprt element(data_plus_offset, element_type);
 
-      c=code_assignt(element, op[2]);
+      code_blockt assert_and_put;
+      if(!disable_runtime_checks)
+      {
+        codet bounds_check=
+          get_array_bounds_check(deref, op[1], i_it->source_location);
+        bounds_check.add_source_location()=i_it->source_location;
+        assert_and_put.move_to_operands(bounds_check);
+      }
+      code_assignt array_put(element, op[2]);
+      array_put.add_source_location()=i_it->source_location;
+      assert_and_put.move_to_operands(array_put);
+      c=std::move(assert_and_put);
```

Comment:
It is an additional step that is not required.
I would prefer to assign the code block to c first and then do all the manipulations on c.
(More instances below).